mantidproject/mantid
# Fix lookup matching from ISIS Reflectometry Preview tab

**Description of work**

**Purpose of work**

This is part two of the fix for the linked issue. This PR adds the functionality to match against lookup rows using both run title and angle, rather than just angle. During testing I also found a crash that occurs when a reduction is triggered without the loaded workspace, so this PR also implements a simple fix for that.



**Summary of work**

This PR moves the logic for extracting the title and theta values from the logs `run_title` entry out of `SearchResult` and into `ParseReflectometryStrings`. This was to allow the logic to be re-used in the `LookupTable` class for finding the title to match against from a `PreviewRow`.

I have also changed the lookup row matching logic in `LookupTable` so that both a `Row` and `PreviewRow` share the same logic and just retrieve the title for matching in a different way.

The full run title is now displayed at the top of the Reduction Preview tab, as requested by our scientists. I have used a label to display this information, as our scientists do not need to edit it and it seemed to give the best behaviour in terms of resizing and word wrap.

Finally I have also included a small change to check if a loaded workspace is available in `PreviewPresenter::runReduction` to fix the crash described in #35825.

**To test:**


- See instructions in the relevant issues.

- For #35704, the angle and title matching can be tested by selecting a region on the instrument view and slice viewer (this will need to be done to get the reduction to complete successfully anyway) and then clicking the `Apply` button at the bottom right of the dialog. This should set the `ROI` and `ROI Detector IDs` values against the correct row in the lookup table on the Experiment Settings tab. To thoroughly test the matching you will need to try different title values - i.e. using the title "seg" in the Experiment Settings lookup row will match the title of run `INTER45455`, while something like "test" will not.
 
Fixes #35704 and fixes #35825. 






---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
